newm-server/src/main/kotlin/io/newm/server/cors/CorsInstall.kt

```diff
@@ -31,3 +51,589 @@ fun Application.installCORS() {
+        allowHeader(HttpHeaders.XHttpMethodOverride)
+    }
+
+    /**
+     * Allows using an origin matching [predicate] for the actual [CORS] request.
+     */
+    public fun allowOrigins(predicate: (String) -> Boolean) {
+        this.originPredicates.add(predicate)
+    }
+
+    /**
+     * Allows using headers prefixed with [headerPrefix] for the actual [CORS] request.
+     */
+    public fun allowHeadersPrefixed(headerPrefix: String) {
+        val prefix = headerPrefix.lowercase()
+        this.headerPredicates.add { name -> name.startsWith(prefix) }
+    }
+
+    /**
+     * Allows using headers matching [predicate] for the actual [CORS] request.
+     */
+    public fun allowHeaders(predicate: (String) -> Boolean) {
+        this.headerPredicates.add(predicate)
+    }
+
+    /**
+     * Allow using a specified [header] for the actual [CORS] request.
+     */
+    public fun allowHeader(header: String) {
+        if (header.equals(HttpHeaders.ContentType, ignoreCase = true)) {
+            allowNonSimpleContentTypes = true
+            return
+        }
+
+        if (header !in CorsSimpleRequestHeaders) {
+            headers.add(header)
+        }
+    }
+
+    /**
+     * Adds a specified [method] to a list of methods allowed by [CORS].
+     *
+     * Note that CORS operates with real HTTP methods only and
+     * doesn't handle method overridden by `X-Http-Method-Override`.
+     */
+    public fun allowMethod(method: HttpMethod) {
+        if (method !in CorsDefaultMethods) {
+            methods.add(method)
+        }
+    }
+}
+
+internal fun PluginBuilder<CORSConfig>.buildPlugin() {
+    val numberRegex = "[0-9]+".toRegex()
+    val allowSameOrigin: Boolean = pluginConfig.allowSameOrigin
+    val allowsAnyHost: Boolean = "*" in pluginConfig.hosts
+    val allowCredentials: Boolean = pluginConfig.allowCredentials
+    val allHeaders: Set<String> =
+        (pluginConfig.headers + CORSConfig.CorsSimpleRequestHeaders).let { headers ->
+            if (pluginConfig.allowNonSimpleContentTypes) headers else headers.minus(HttpHeaders.ContentType)
+        }
+    val originPredicates: List<(String) -> Boolean> = pluginConfig.originPredicates
+    val headerPredicates: List<(String) -> Boolean> = pluginConfig.headerPredicates
+    val methods: Set<HttpMethod> = HashSet(pluginConfig.methods + CORSConfig.CorsDefaultMethods)
+    val allHeadersSet: Set<String> = allHeaders.map { it.toLowerCasePreservingASCIIRules() }.toSet()
+    val allowNonSimpleContentTypes: Boolean = pluginConfig.allowNonSimpleContentTypes
+    val headersList = pluginConfig.headers
+        .filterNot { it in CORSConfig.CorsSimpleRequestHeaders }
+        .let { if (allowNonSimpleContentTypes) it + HttpHeaders.ContentType else it }
+    val methodsListHeaderValue = methods
+// *** Ensure that we send ALL methods in the response, not just the ones that are not in the default set
+//        .filterNot { it in CORSConfig.CorsDefaultMethods }
```

Comment:
This is the line preventing the server from telling the client that POST requests are among the allowed methods. We'll see if this allows things to succeed.  All this code was just copied from KTOR so I could tweak this for a test.